src/ceph-disk/ceph_disk/main.py

```diff
@@ -2123,20 +2179,89 @@ def __init__(self, args):
         super(PrepareBluestoreBlock, self).__init__(args)
 
     def get_space_size(self):
-        return 0  # get as much space as possible
+        block_size = get_conf(
+            cluster = self.args.cluster,
+            variable = 'bluestore_block_size',
+        )
 
-    def desired_partition_number(self):
-        if self.args.block == self.args.data:
-            num = 2
+        if block_size is None:
+            return 0  # get as much space as possible
         else:
-            num = 0
-        return num
+            return int(block_size) / 1048576 # MB
+
+    def desired_partition_number(self):
+        # FIXME: use a fixed partition number for now
```

Comment:
> we should set num = 0 for block{,.wal,.db} partition to let it pick first available partition. 

That sound right. How about this: for all block block.wal block.db, if it is the same as args.data then use a fixed partition number. If it's different use num = 0 so that it picks the next available partition.